File: backend/src/main/java/fu/osms/catalog/service/CategoryService.java
package fu.osms.catalog.service;

import fu.osms.catalog.dto.request.CategoryRequest;
import fu.osms.catalog.dto.response.CategoryDashboardResponse;
import fu.osms.catalog.dto.response.CategoryNodeResponse;
import fu.osms.catalog.dto.response.CategoryResponse;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.UUID;

public interface CategoryService {

    CategoryResponse create(CategoryRequest request);

    CategoryResponse getById(UUID id);

    List<CategoryResponse> getAll();

    List<CategoryResponse> getRootCategories();

    List<CategoryResponse> getSubCategories(UUID parentId);

    CategoryResponse update(UUID id, CategoryRequest request);

    void delete(UUID id);

    public List<CategoryNodeResponse> getCategoryTree();

    public CategoryDashboardResponse getCategoryDashboardData(Pageable pageable);

}

File: frontend/src/features/catalog/services/categoryService.js
import axiosClient from '../../../api/axiosClient';

const categoryService = {
    getAll: async () => {
        const response = await axiosClient.get('/categories');
        const data = response.data?.data || response.data || response;
        if (Array.isArray(data)) return data;
        if (data.content) return data.content;
        if (Array.isArray(data.data)) return data.data;
        return [];
    },
};

export default categoryService;
